### PR TITLE
setup1-contributor enable password to be supplied via command line argument

### PR DESCRIPTION
**Pull request by kellpossible**
+ Refactor setup1-contributor command line interface to use `structopt` instead of custom `CLI` trait, much simpler, and more type-safe.
+ Allow setup1-contributor's `passphrase` to be specified via an optional command line argument for both the `generate` and `contribute` subcommands.

### CODE REVIEW

File: setup1-contributor/src/cli/commands/contribute.rs [hunk 1]
```diff
@@ -1,107 +1,51 @@
...
+    about = "Contribute to Aleo Setup I",
+    rename_all = "snake-case",
+    setting(AppSettings::ColoredHelp),
+    setting(AppSettings::DisableHelpSubcommand),
+    setting(AppSettings::DisableVersion)
+)]
+pub struct ContributeOptions {
+    /// Specify how the responses are uploaded.
+    #[structopt(
+        name("upload_mode"),
```
Comment (kellpossible) [inline setup1-contributor/src/cli/commands/contribute.rs:23]: This no longer needs to be manually renamed

File: setup1-contributor/src/utils.rs [hunk 2]
```diff
@@ -170,20 +170,67 @@ pub fn sign_contribution_state<R: Rng + CryptoRng>(
...
-pub fn upload_mode_from_str(upload_mode: &str) -> Result<UploadMode> {
-    match upload_mode {
-        "auto" => Ok(UploadMode::Auto),
-        #[cfg(feature = "azure")]
-        "azure" => Ok(UploadMode::Azure),
-        "direct" => Ok(UploadMode::Direct),
-        _ => Err(UtilsError::UnknownUploadModeError(upload_mode.to_string()).into()),
+impl UploadMode {
+    pub fn variants() -> &'static [&'static str] {
+        &["auto", "direct"]
```
Comment (ibaryshnikov) [inline setup1-contributor/src/utils.rs:220]: azure?

**Comment by ibaryshnikov** [comment]
As you are making this change, what if we drop positional arguments? It was one of the most confusing things for me at first: what is a first argument? What is a second? Having key-value options seems way more transparent and self-documenting. For example the following command:
`./aleo-setup-contributor generate --keys-path keys.json`
seems way more obvious than
`./aleo-setup-contributor generate keys.json`
and it provides quite a nice help message.
Also, what is a more common thing in cli tools: snake_case options or kebab-case options? By default structopt uses kebab-case, which seems pretty natural for linux cli tools. Check `ls --help` for an example

**Comment by apruden2008** [comment]
> As you are making this change, what if we drop positional arguments? It was one of the most confusing things for me at first: what is a first argument? What is a second? Having key-value options seems way more transparent and self-documenting. For example the following command:
> `./aleo-setup-contributor generate --keys-path keys.json`
> seems way more obvious than
> `./aleo-setup-contributor generate keys.json`
> and it provides quite a nice help message.
> Also, what is a more common thing in cli tools: snake_case options or kebab-case options? By default structopt uses kebab-case, which seems pretty natural for linux cli tools. Check `ls --help` for an example

I think that sounds good, but is probably a separate PR

**Comment by kellpossible** [comment]
> As you are making this change, what if we drop positional arguments? It was one of the most confusing things for me at first: what is a first argument? What is a second? Having key-value options seems way more transparent and self-documenting. For example the following command:
> `./aleo-setup-contributor generate --keys-path keys.json`
> seems way more obvious than
> `./aleo-setup-contributor generate keys.json`
> and it provides quite a nice help message.
> Also, what is a more common thing in cli tools: snake_case options or kebab-case options? By default structopt uses kebab-case, which seems pretty natural for linux cli tools. Check `ls --help` for an example

Thanks @ibaryshnikov, I really like all these suggestions, but I think we can make a separate issue/PR for them, as I was trying to match compatibility with the existing CLI interface with this PR

**Review by apruden2008 — APPROVED** [review]

**Review by ibaryshnikov — APPROVED** [review]

**Pull request merged**